ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockCreator.java

```diff
@@ -263,15 +263,13 @@ private ProcessableBlockHeader createPendingBlockHeader(final long timestamp) {
       final EIP1559 eip1559 = protocolSpec.getEip1559().orElseThrow();
       if (eip1559.isForkBlock(newBlockNumber)) {
         gasLimit = gasLimit * eip1559.getFeeMarket().getSlackCoefficient();
-        baseFee = eip1559.getFeeMarket().getInitialBasefee();
-      } else {
-        baseFee =
-            eip1559.computeBaseFee(
-                newBlockNumber,
-                parentHeader.getBaseFee().orElseThrow(),
-                parentHeader.getGasUsed(),
-                eip1559.targetGasUsed(parentHeader));
       }
+      baseFee =
+          eip1559.computeBaseFee(
+              newBlockNumber,
+              parentHeader.getBaseFee().orElseThrow(),
```

Comment:
Maybe I'm missing something but if we're in the fork block. The parent has no baseFee and this will throw an exception ? 🤔

Reply:
Forgot about the fork block thing  ☝️